src/serac/physics/state/finite_element_dual.hpp

```diff
@@ -56,9 +83,44 @@ class FiniteElementDual : public FiniteElementVector {
    *
    * @see <a href="https://mfem.org/pri-dual-vec/">MFEM documentation</a> for details
    *
+   * @note It is more mathematically correct to use a linear form on the dual space. However, grid functions
+   * are common containers in MFEM-based codes, so this is left for convenience.
+   *
    * @param grid_function The grid function used to initialize the underlying true vector.
    */
   void setFromGridFunction(const mfem::ParGridFunction& grid_function) { grid_function.ParallelAssemble(*this); }
+
+  /**
+   * @brief Fill a user-provided linear form based on the underlying true vector
+   *
+   * This distributes true vector dofs to the finite element (local) dofs by multiplying the true dofs
+   * by the restriction transpose operator.
+   *
+   * @see <a href="https://mfem.org/pri-dual-vec/">MFEM documentation</a> for details
+   *
+   * @param linear_form The linear form used to initialize the underlying true vector.
+   */
+  void fillLinearForm(mfem::ParLinearForm& linear_form) const
```

Comment:
I think it's a great addition and much more clear to have the `fillLinearForm` and `setFromLinearForm` methods for `FiniteElementDual`. Are the `gridfunction` version of these methods still necessary? It seems confusing to a user to support both methods

Reply:
Agreed. I'm hesitant to delete them though because people tend to use grid functions for everything. If LiDO is OK with me deleting, I'm OK with it as it does make more mathematical sense. Any thoughts @sethwatts ?

Reply:
I would go farther and actually prefer that you delete the `GridFunction` filler due to the mathematical inconsistency.

Reply:
Question: Would you ever want to set a dual vector from a coefficient?

Reply:
I'm not coming up with any examples where we'd want to do that. @kswartz92 can you think of any? The closest I can think of is of a load vector, but I think the integration of a traction field into a load linear form is something we wouldn't want to skip.

Reply:
I don't think I've ever set a dual vector from a coefficient